### PR TITLE
HomeStiling => stilized activity list, activity details and create activity

### PR DESCRIPTION
**Pull request by aghermanHP**
HomeStiling => cange group card padding, maked responsive, chanched color of cards and aded info, @nlaptedulce please verify

### CODE REVIEW

File: src/components/activity/VActivity.vue [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+</template>
+
+<script>
+  import ActivityService from '../../services/ActivityApi';
+
+
+  export default {
+    data () {
+      return {
+        ActivityDetails: {
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue]: rename this object, like activity

File: src/components/activity/VActivity.vue [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          id: null,
+          name: '',
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue]: remove all variables from this object

File: src/components/activity/VActivity.vue [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+          public: true,
+        }
+
+      }
+
+    },
+      created () {
+        let idActivity = localStorage.getItem('idActivity');
+        ActivityService.getActivityDetails(idActivity).then ((response) => {
+          console.log (response);
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue]: remove console.log

File: src/components/activity/VActivity.vue [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            id: null
+          }],
+          public: true,
+        }
+
+      }
+
+    },
+      created () {
+        let idActivity = localStorage.getItem('idActivity');
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue]: get id from path of page

File: src/components/activity/VActivity.vue [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+          console.log (response);
+          this.ActivityDetails = response.data;
+        })
+          .catch (error => {
+            console.log (error)
+          })
+      },
+   
+  }
+</script>
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue:107]: add a line after script

File: src/components/auth/Login.vue [hunk 6]
```diff
@@ -79,26 +79,27 @@
 <script>
 import LoginService from '../../services/loginApi';
```
Comment (nlaptedulce) [inline src/components/auth/Login.vue:80]: add a line after import

File: src/components/auth/Login.vue [hunk 7]
```diff
@@ -79,26 +79,27 @@
...
 export default {
-  data() {
+  data () {
     return {
       form: {
         username: '',
         password: '',
       },
+      name: '',
```
Comment (nlaptedulce) [inline src/components/auth/Login.vue]: remove name and JWT

File: package.json [hunk 8]
```diff
@@ -4,8 +4,7 @@
   "private": true,
   "scripts": {
     "serve": "vue-cli-service serve",
-    "build": "vue-cli-service build",
-    "lint": "vue-cli-service lint"
```
Comment (nlaptedulce) [inline package.json]: add eslint

File: src/main.js [hunk 9]
```diff
@@ -3,11 +3,13 @@ import App from './App.vue';
 import router from './route';
 import store from './store';
 import './plugins';
+import CommonServices from './services/CommunServices';
```
Comment (nlaptedulce) [inline src/main.js]: remove from main.js this communServices

File: src/services/CommunServices.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios';
```
Comment (nlaptedulce) [inline src/services/CommunServices.js]: change name of axios in Axios

Comment (nlaptedulce) [inline src/services/CommunServices.js]: name of service must be singular

File: src/services/loginApi.js [hunk 11]
```diff
@@ -1,9 +1,10 @@
 import Axios from 'axios';
-const API_LOGIN_RESOURCE = 'http://api.pentabee.local/api/login';
+const token = localStorage.getItem ('token');
+import CommonServices from './CommunServices';
 
 export default {
-
   login(data) {
-    return Axios.post(API_LOGIN_RESOURCE, data);
+    CommonServices.getToken();
```
Comment (nlaptedulce) [inline src/services/loginApi.js]: you don't need here token

File: src/views/Home.vue [hunk 12]
```diff
@@ -1,60 +1,75 @@
...
+
+<script>
+  import ActivityService from '../services/ActivityApi';
+  export default {
+    name: 'ActivityList',
+    data () {
+      return {
+        page: 1,
+        results: [],
+        numResults: null,
```
Comment (nlaptedulce) [inline src/views/Home.vue]: you don't need all this variables

File: src/views/Home.vue [hunk 13]
```diff
@@ -1,60 +1,75 @@
...
+        perPage: null,
+        numPages: null
+      }
+    },
+    mounted(){
+      ActivityService.getActivityList (this.page).then ((response) => {
+        console.log (response.data.results);
+        this.results = response.data.results;
+        console.log('number of items'+ this.results.length);
```
Comment (nlaptedulce) [inline src/views/Home.vue]: remove console logs

File: src/views/Home.vue [hunk 14]
```diff
@@ -1,60 +1,75 @@
...
+      })
+        .catch (error => {
+          console.log (error)
+        })
+    },
+    methods: {
+      getInfo (id) {
+        localStorage.setItem('idActivity',id);
```
Comment (nlaptedulce) [inline src/views/Home.vue]: remove setItem

File: src/route/router.js [hunk 15]
```diff
@@ -26,10 +27,10 @@ export default new Router({
       component: Registration,
     },
     {
-      path: '/profile',
-      name: 'profile',
-      component: Profile,
-    },
+      path: ':AId/activity',
```
Comment (nlaptedulce) [inline src/route/router.js]: just id

File: src/components/activity/VActivity.vue (hunk 1 above)
Comment (aghermanHP) [reply src/components/activity/VActivity.vue]: renamed :)

File: src/components/activity/VActivity.vue (hunk 2 above)
Comment (aghermanHP) [reply src/components/activity/VActivity.vue]: done

File: src/components/activity/VActivity.vue (hunk 3 above)
Comment (aghermanHP) [reply src/components/activity/VActivity.vue]: all redy

File: src/components/activity/VActivity.vue (hunk 4 above)
Comment (aghermanHP) [reply src/components/activity/VActivity.vue]: done

File: src/components/activity/VActivity.vue (hunk 5 above)
Comment (aghermanHP) [reply src/components/activity/VActivity.vue:107]: done

File: src/services/loginApi.js (hunk 11 above)
Comment (aghermanHP) [reply src/services/loginApi.js]: done

File: src/views/Home.vue (hunk 12 above)
Comment (aghermanHP) [reply src/views/Home.vue]: done

File: src/services/loginApi.js (hunk 11 above)
Comment (aghermanHP) [reply src/services/loginApi.js]: ok

File: src/views/Home.vue (hunk 12 above)
Comment (aghermanHP) [reply src/views/Home.vue]: done

File: src/views/Home.vue (hunk 13 above)
Comment (aghermanHP) [reply src/views/Home.vue]: done

File: src/views/Home.vue (hunk 14 above)
Comment (aghermanHP) [reply src/views/Home.vue]: done

File: src/route/router.js (hunk 15 above)
Comment (aghermanHP) [reply src/route/router.js]: ok

File: src/App.vue [hunk 16]
```diff
@@ -20,12 +20,13 @@ import VFooter from '@/views/TheFooter';
 import VSidebar from '@/components/VSidebar';
 import VLayout from '@/components/VLayout';
 
+
```
Comment (nlaptedulce) [inline src/App.vue]: remove, please, this line

File: src/components/VSidebar.vue [hunk 17]
```diff
@@ -8,35 +8,41 @@ import { mapActions } from 'vuex';
...
       {
         href: '/charts',
-        title: 'List',
+        title: 'Activity',
         icon: 'fa fa-cog',
         child: [
           {
-            href: '/charts/sublink',
-            title: 'Sub Link',
+            href: '/ActivityList',
```
Comment (nlaptedulce) [inline src/components/VSidebar.vue]: routes are name with: kebab-case

File: src/components/activity/VActivity.vue [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+<script>
+  import ActivityService from '../../services/ActivityApi';
+  import CommonServices from '../../services/Services'
+
+  export default {
+
+    data () {
+      return {
+        Activity: {
+          owner: []
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue]: owner should be object

File: src/components/activity/VActivity.vue [hunk 19]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        idUser: null
+      }
+    },
+
+    created () {
+      ActivityService.getActivityDetails (this.$route.params.ActivityId).then ((response) => {
+        this.Activity = response.data;
+        this.idUser = CommonServices.idUser;
+
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue]: remove these lines

File: src/route/router.js [hunk 20]
```diff
@@ -26,10 +28,20 @@ export default new Router({
...
     {
-      path: '/profile',
-      name: 'profile',
-      component: Profile,
+      path: '/ActivityList',
+      name: 'activityList',
+      component: ActivityList,
+    },
+    {
+      path: '/activity/:ActivityId',
```
Comment (nlaptedulce) [inline src/route/router.js]: rename param - ActivityId - as "activityId"

File: src/route/router.js [hunk 21]
```diff
@@ -26,10 +28,20 @@ export default new Router({
...
+      name: 'activity',
+      component: Activity,
     },
+    {
+      path: '/activityCreate',
```
Comment (nlaptedulce) [inline src/route/router.js]: kebab-case

File: src/services/ActivityApi.js [hunk 22]
```diff
@@ -0,0 +1,22 @@
+import Axios from 'axios';
+import CommonServices from './Services';
+
+export default {
+  getActivityDetails (id) {
+    CommonServices.getToken();
+    return Axios.get(CommonServices.base_url+ 'activities/'+ id);
```
Comment (nlaptedulce) [inline src/services/ActivityApi.js]: before and after + should be space

File: src/services/ActivityApi.js [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  deleteActivity(id){
+    CommonServices.getToken();
+    return Axios.delete(CommonServices.base_url+'activities/' + id +'/delete')
+  },
+  createActivity(data){
+    CommonServices.getToken();
+    return Axios.post(CommonServices.base_url+'activities/create', data)
+  }
+
```
Comment (nlaptedulce) [inline src/services/ActivityApi.js]: remove this line

File: src/services/Services.js [hunk 24]
```diff
@@ -0,0 +1,17 @@
+import Axios from 'axios';
```
Comment (nlaptedulce) [inline src/services/Services.js]: rename this service as TokenService

File: src/services/Services.js [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+import jwt_decode from 'jwt-decode';
```
Comment (nlaptedulce) [inline src/services/Services.js:2]: after import a line

File: src/services/Services.js [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+const USERNAME =  decoded.username;
+
+export default {
+  base_url: BASE_URL,
+  token: TOKEN,
+  idUser: IdUSER,
+  username: USERNAME,
+  getToken() {
+   return  Axios.defaults.headers.common['Authorization'] = `Bearer ${TOKEN}`;
+  },
```
Comment (nlaptedulce) [inline src/services/Services.js]: remove this comma

File: src/services/Services.js [hunk 27]
```diff
@@ -0,0 +1,17 @@
+import Axios from 'axios';
+import jwt_decode from 'jwt-decode';
+const TOKEN = localStorage.getItem ('token');
+const BASE_URL='http://api.pentabee.local/api/';
+const decoded = jwt_decode(TOKEN);
+const IdUSER = decoded.id;
```
Comment (nlaptedulce) [inline src/services/Services.js]: All constants should be named with snake_case an uppercase

File: src/services/Services.js [hunk 28]
```diff
@@ -0,0 +1,17 @@
...
+const USERNAME =  decoded.username;
+
+export default {
+  base_url: BASE_URL,
```
Comment (nlaptedulce) [inline src/services/Services.js]: variables should be with camelCase

File: src/views/TheHeader.vue [hunk 29]
```diff
@@ -63,3 +63,18 @@
...
   }
 </style>
+<script>
+  export default {
+    data(){
+      return{
+      
+      }
+    },
+    methods:{
```
Comment (nlaptedulce) [inline src/views/TheHeader.vue]: space before {
revises the spaces and arranges the code

File: src/App.vue (hunk 16 above)
Comment (aghermanHP) [reply src/App.vue]: ok

File: src/components/activity/VActivity.vue [hunk 30]
```diff
@@ -0,0 +1,164 @@
...
+    font-size: initial;
+    padding: 0;
+  }
+  
+  .deadline {
+    color: red;
+    font-size: initial;
+    padding: 0;
+  }
+
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue]: remove this line

File: src/main.js [hunk 31]
```diff
@@ -3,9 +3,27 @@ import App from './App.vue';
 import router from './route';
 import store from './store';
 import './plugins';
-
+import Datetime from 'vue-datetime';
+import './assets/styles/vueDataTime.css'; //style for datetime piker
```
Comment (nlaptedulce) [inline src/main.js:7]: you can import this file in main.scss

File: src/components/VSidebar.vue (hunk 17 above)
Comment (aghermanHP) [reply src/components/VSidebar.vue]: done

File: src/components/activity/VActivity.vue (hunk 18 above)
Comment (aghermanHP) [reply src/components/activity/VActivity.vue]: ok

File: src/services/ActivityApi.js (hunk 22 above)
Comment (aghermanHP) [reply src/services/ActivityApi.js]: :+1:  +  :)

File: src/services/ActivityApi.js (hunk 23 above)
Comment (aghermanHP) [reply src/services/ActivityApi.js]: ok

File: src/components/VSidebar.vue (hunk 17 above)
Comment (aghermanHP) [reply src/components/VSidebar.vue]: done

File: src/components/activity/VActivity.vue (hunk 18 above)
Comment (aghermanHP) [reply src/components/activity/VActivity.vue]: ok

File: src/components/activity/VActivity.vue (hunk 19 above)
Comment (aghermanHP) [reply src/components/activity/VActivity.vue]: ok

File: src/route/router.js (hunk 20 above)
Comment (aghermanHP) [reply src/route/router.js]: done

File: src/services/Services.js (hunk 24 above)
Comment (aghermanHP) [reply src/services/Services.js]: done

File: src/services/Services.js (hunk 25 above)
Comment (aghermanHP) [reply src/services/Services.js:2]: ok

File: src/services/Services.js (hunk 26 above)
Comment (aghermanHP) [reply src/services/Services.js]: ok

File: src/services/Services.js (hunk 24 above)
Comment (aghermanHP) [reply src/services/Services.js]: OK

File: src/services/Services.js (hunk 25 above)
Comment (aghermanHP) [reply src/services/Services.js:2]: OK

File: src/services/Services.js (hunk 26 above)
Comment (aghermanHP) [reply src/services/Services.js]: OK

File: src/services/Services.js (hunk 27 above)
Comment (aghermanHP) [reply src/services/Services.js]: OK

File: src/services/Services.js (hunk 28 above)
Comment (aghermanHP) [reply src/services/Services.js]: ok

File: src/views/TheHeader.vue (hunk 29 above)
Comment (aghermanHP) [reply src/views/TheHeader.vue]: ok

File: src/components/activity/VActivity.vue (hunk 30 above)
Comment (aghermanHP) [reply src/components/activity/VActivity.vue]: done

File: src/main.js (hunk 31 above)
Comment (aghermanHP) [reply src/main.js:7]: done

File: src/components/activity/VActivity.vue (hunk 19 above)
Comment (aghermanHP) [reply src/components/activity/VActivity.vue]: ok

File: src/main.js (hunk 31 above)
Comment (aghermanHP) [reply src/main.js:7]: i c'ant becose it is .css but main is .scss

**Pull request merged**